rasa/shared/core/trackers.py

```diff
@@ -813,3 +828,56 @@ def is_prev_action_listen_in_state(state: State) -> bool:
     """
     prev_action_name = state.get(PREVIOUS_ACTION, {}).get(ACTION_NAME)
     return prev_action_name == ACTION_LISTEN_NAME
+
+
+def get_trackers_for_conversation_sessions(
```

Comment:
Can we consolidate this one with [_split_conversation_at_restarts](https://github.com/RasaHQ/rasa/blob/cee30cbacf1655f4796d49d15a9ad56a6ec2dc5e/rasa/core/training/interactive.py#L747)?

Reply:
I don't see the benefit in trying to combine the two - the one in `interactive.py` works on serialised events, this one doesn't

Reply:
I don't think we should duplicate the logic to split conversations. E.g. we could make the one in `interactive` work on serialized events, too. It's not ideal for performance, but it will be called just once, so it's fine.

Reply:
but the split is a very different one. in interactive we split on the `restart` event - here we split on `ActionExecuted(action_session_start)`. All of this could be parametrised in the function but I don't see the benefit in doing that, especially since we'd also have to deserialise events in the interactive case

Reply:
Okay I've just convinced myself 😄   I've moved the split